scripts/bootstrap-ansible.sh

```diff
@@ -64,8 +64,19 @@ pushd ${OA_DIR}
     exit 99
   fi
 
-  # Now use GROUP_VARS of OSA and RPC
-  sed -i "s|GROUP_VARS_PATH=.*|GROUP_VARS_PATH=\"\${GROUP_VARS_PATH:-${OA_DIR}/playbooks/inventory/group_vars/:${BASE_DIR}/group_vars/:/etc/openstack_deploy/group_vars/}\"|" /usr/local/bin/openstack-ansible.rc
-  sed -i "s|HOST_VARS_PATH=.*|HOST_VARS_PATH=\"\${HOST_VARS_PATH:-${OA_DIR}/playbooks/inventory/host_vars/:${BASE_DIR}/host_vars/:/etc/openstack_deploy/host_vars/}\"|" /usr/local/bin/openstack-ansible.rc
+  # RPC-O has roles in its own git tree, so we need to add it to the
+  # path for Ansible to search.
+  sed -i "s|/etc/ansible/roles:|/etc/ansible/roles:${RPCD_DIR}/playbooks/roles:|" /usr/local/bin/openstack-ansible.rc
+
+  # NOTE(cloudnull): Sync {group,host}_vars from rpc-o into the proper OSA
+  #                  directory which supports the default override interface.
+  if [[ -d "${BASE_DIR}/group_vars" ]];then
+    rsync -av ${BASE_DIR}/group_vars /etc/openstack_deploy/
+    var_perms "/etc/openstack_deploy/group_vars"
```

Comment:
Instead of nesting a function here (considering the objective is to decrease the layers) you could also just do: ``chmod ugo+rX <path>``

Reply:
I'm not sure we should have the yaml files be executable. I mean it shouldn't cause any harm but there's no reason to set the execution bit.

Reply:
Capital X would only set the executable bit on directories, not files.